### ISSUE TITLE
Training with self-made dataset

### ISSUE THREAD

**Issue by ShaneZHOU1994** [NVlabs/ssn_superpixels#12]
hello, very thankful for sharing the whole program. My question is that I want to train this SSN with my own dataset, which basically consists of original images from BSDS500 training images to be input, and their superpixel segmentation data (by some classical superpixel algorithm) to be ground truth. 
So could you provide some documentation explaining how to plug in user defined training dataset to replace BSDS500 training set ? thank you very much

**Comment by varunjampani** [comment]
For training SSN with BSDS, we convert GT segmentation into one-hot encoding and we assume that there are maximum 50 segments in an image. See 'convert_label' function (lines 59-74) in fetch_and_transform_data.py. You would need to modify this function if you want to train with existing superpixels as GT. You may face memory issues if this one-hot encoding matrix is big.

**Comment by ShaneZHOU1994** [comment]
Hello,
I still meet this problem. Here is the detail:
	I notice that the structure of segmentation data in BSDS500 is rather complicated (those .mat files), which contain lots of information but not useful to my study; as my concern, the only important info is the matrix of marked labels (481x321 size, with integers marking each pixel to its segmented region). So my data (also .mat files) just contain this matrix.
	In order to input my ground truth correctly, here is how I modify the code:
1. in function “fetch_and_transform_data” and “fetch_and_transform_patch_data”, I comment these two lines  
`# t = np.random.randint(0, len(gtseg_all['groundTruth'][0]))`
`# gtseg = gtseg_all['groundTruth'][0][t][0][0][0]` 
and write mine:` gtseg = gtseg_all['groundTruth'] ` to match my data structure;
2. also, I followed your instruction to change the max number of segments from 50 to 120, (actually this number 50 appears in five places in whole program and they seem to refer to the same thing as this max number);
3. the result is that I get stuck when it create input layer, with no error message, just stay stuck and still, after showing this line:
`Outputs: [‘img’, ‘spixel_init’, ‘feat_spixel_init’, ‘label’, ‘problabel’]`

now I’m thinking  this problem might come from the structure of GT data, because  the only thing I changed is the replacement of GT data. So could you tell me what you think about my problem and explain a bit more about how you treat the structure of GT data (‘from .mat file’) , especially the label matrix? Did I miss some very important things for the processing of GT data?

Thanks

**Comment by varunjampani** [comment]
Your understanding of gt segmentations is correct. I just read label integers from mat files and convert them into one-hot encodings. It is difficult to debug without error message. You may need to debug this manually with 'pdb' stop statements 'pdb.set_trace()'.

**Comment by ShaneZHOU1994** [comment]
Thanks for your advice, my problem is now solved (after long time of debugging and adjusting parameters in the code, finally...)
